statsmodels/nonparametric/smoothers_lowess.py

```diff
@@ -150,6 +157,18 @@ def lowess(endog, exog, frac=2.0/3.0, it=3, delta=0.0, xvals=None, is_sorted=Fal
     if endog.shape[0] != exog.shape[0] :
         raise ValueError('exog and endog must have same length')
 
+    if windows is not None:
+        if windows.ndim != 2:
+            raise ValueError('windows must be a 2D matrix')
+        if windows.shape[1] != 2:
+            raise ValueError('second dim of windows must be 2')
+        if given_xvals:
+            raise ValueError('windows with xvals is not implemented')
+        if delta != 0.0:
+            raise ValueError('windows with delta is not implemented')
+        if windows is not None and windows.shape[0] != xvals.shape[0]:
```

Comment:
AFAICS,
xvals can be None here

`xvals = exog` is assigned on line 199

Reply:
yes. this PR is not clean yet. I would make a proper PR if there is an interest in this kind of extension.